R/entities/abstract-dataset.R

```diff
@@ -21,37 +23,53 @@ AbstractDataset <- R6::R6Class("AbstractDataset",
                                                                 #' @description
                                                                 #' Create an abstract dataset - you should only do this through a child extension
                                                                 #' @param name String name of dataset
                                                                 #' @param publication_metadata PublicationMetadata object defining the publication metadata
                                                                 #' @param region_scale string specifying region scale. Currently Region|State|Country
+                                                                #' @param data Function for downloading data
                                                                 #' @param case_modifier lambda for modifying cases. must accept cases as a single parameter and return it
                                                                 #' @param generation_time String, optional override line list file
                                                                 #' @param incubation_period String, optional override line list file
                                                                 #' @param reporting_delay String, optional override line list file
                                                                 #' @param stable Boolean
+                                                                #' @param data_window Number of weeks to use when estimating (referenced from most recent date)
                                                                 #' @param data_args list for passing to getregionaldata as arguments
                                                                 #' @param truncation Integer Number of days to trim off the end of the newest end of the data
+                                                                #' @param regional_epinow_opts List of options to be passed to regional_epinow
                                                                 #' @return A new `AbstractDataset` Object
                                  initialize = function(
                                    name,
                                    publication_metadata,
                                    region_scale,
+                                   data = NA,
                                    case_modifier = NA,
                                    generation_time = NA,
                                    incubation_period = NA,
                                    reporting_delay = NA,
                                    stable = TRUE,
                                    data_args = NULL,
-                                   truncation = 0
+                                   truncation = 0,
+                                   data_window = 12,
+                                   regional_epinow_opts = list(rt = rt_opts(prior = list(mean = 1, sd = 0.2)),
```

Comment:
These extra values need saving below to the class or else they won't do nout!
```R
self$regional_epinow_opts = regional_epinow_opts
```

Reply:
done